src/signal.c

```diff
@@ -15,113 +16,162 @@
 
 #include <libswiftnav/signal.h>
 
-typedef struct {
-  u8 constellation;
-  u8 band;
-  u16 sat_count;
-  u16 (*sat_get)(u32 local_index);
-  u32 (*local_index_get)(u16 sat);
```

Comment:
FYI the idea with the local_index_get() stuff was that we could have non-contiguous blocks of sat values for a given code. But I think we should keep your changes until we need something more sophisticated (if that ever happens).